### PR TITLE
UI Components Add-On

### PR DESCRIPTION
**Pull request by nummi**
The goal of this PR is to move as much generic UI logic to another location and keep the main repo for _business logic_. This is a huge commit so I'll leave comments where I think they will be useful.

Step two could include: more tests and documentation, usage of a component playground like _Storybook_ or _Ember CLI Addon Docs_.

This will also make life easier if we go down the _Inspector Add-ons_ route.

### CODE REVIEW

**Comment by nummi** [comment]
_explanation coming soon..._

**Comment by rwwagner90** [comment]
This is awesome work! I'll give it a thorough review this week at some point. Pretty swamped with some other things currently.

File: app/styles/app.scss [hunk 1]
```diff
@@ -25,22 +25,9 @@
...
-@import "nav";
-@import "notice";
 @import "object_inspector";
-@import "pill";
 @import "route_tree";
-@import "send_to_console";
-@import "sidebar_toggle";
-@import "split";
-@import "toolbar";
-@import "warning";
```
Comment (nummi) [inline app/styles/app.scss]: Most of these were moved to the add-on — a few were no longer being used.

File: app/styles/component_tree.scss [hunk 2]
```diff
@@ -70,27 +70,6 @@
...
-  svg {
-    transition-duration: 0.1s;
-    transition-property: transform;
-  }
-
-  &.collapsed svg {
-    transform: rotate(-90deg);
-  }
-}
-
```
Comment (nummi) [inline app/styles/component_tree.scss]: All of this was rolled into the `disclosure-triangle` component

File: app/templates/components/view-item.hbs [hunk 3]
```diff
@@ -53,13 +53,13 @@
...
-      {{send-to-console action=sendObjectToConsole param=model.value.objectId}}
+      {{ui/send-to-console action=sendObjectToConsole param=model.value.objectId}}
     </div>
   {{else}}
     --
   {{/if}}
 {{/list.cell}}
 
 {{#list.cell class="list__cell_size_small list__cell_value_numeric"}}
-  <span class="pill pill--not-clickable pill--small js-view-duration">{{ms-to-time model.value.duration}}</span>
```
Comment (rwwagner90) [inline app/templates/components/view-item.hbs]: No need for the `not-clickable` class?

File: app/templates/container-type-toolbar.hbs [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
 
   <div class="divider" aria-hidden="true"></div>
 
-  {{send-to-console action=(action "sendContainerToConsole")}}
+  {{ui/send-to-console action=(action "sendContainerToConsole")}}
 
   <div class="divider" aria-hidden="true"></div>
 
   <div
-    class="toolbar__search toolbar__search--small js-container-instance-search"
```
Comment (rwwagner90) [inline app/templates/container-type-toolbar.hbs]: No need for the `toolbar__search--small` class anymore?

File: lib/ui/addon/components/sidebar-toggle.js [hunk 5]
```diff
@@ -1,21 +1,14 @@
...
 
+export default Component.extend({
+  layout,
   tagName: 'button',
-
-  side: 'right',
-
-  isExpanded: false,
-
-  isRight: equal('side', 'right'),
```
Comment (rwwagner90) [inline lib/ui/addon/components/sidebar-toggle.js]: No need for this right stuff anymore?

Comment (nummi) [reply lib/ui/addon/components/sidebar-toggle.js]: As far as I can tell we never had a sidebar toggle for the left sidebar. We can add this back if we ever need it.

File: app/templates/container-type-toolbar.hbs (hunk 4 above)
Comment (nummi) [reply app/templates/container-type-toolbar.hbs]: It only made the search smaller by `25px`. ¯\_(ツ)_/¯

File: app/templates/components/view-item.hbs (hunk 3 above)
Comment (nummi) [reply app/templates/components/view-item.hbs]: It didn't seen to have any effect. The class name didn't really describe what it did anyway. Also, I just changed the base `.pill` class to have that "not-clickable" functionality which is just inheriting the parents cursor.

**Pull request merged**